### PR TITLE
fix(pluginmanager): resolve dtm init prompt error during first download

### LINKED ISSUE #654: :bomb: `Bug`: `dtm init` prompt error during first download
### What Happened?

Even if the plugin is downloaded for the first time, `dtm` still prompts "Plugin: %s already exists, no need to download.".

image

There is a small logic bug in the code.

Related Code:
https://github.com/devstream-io/devstream/blob/8023fa202cade666d8a1f5b5a4a6c936e6eb19a3/internal/pkg/pluginmanager/manager.go#L20-L84

### How to Reproduce?

Execute `dtm init` and download the previously undownloaded plugins.

I suggest you use `release 0.6.1` instead of `release 0.6.0`.  Also, you can build dtm local.

### How to fix it?

**It's enough to just adjust `if` and `log` logic.**

### Anything else

**I think it's a `good first issue`.**

Note:      
* [Development Workflow](https://docs.devstream.io/en/latest/development/development-workflow/)
* The result of this bug fix is currently difficult to test. Because:
> If your dtm binary's MD5 sum doesn't match the MD5 sum from our release page, dtm init will stop. If your local dtm MD5 differs, it indicates that you built the binary yourself (for developing purposes, for example). Due to the nature of the Go plugin, dtm must be built together with the corresponding plugins. So, if you are building dtm, you should also build the plugins as well, in which case, you do not need to run dtm init to download the plugins.

### DevStream Version

latest

### PR DESCRIPTION
**Pull request by azhsmesos**
Signed-off-by: zhaozhenhang zhaozhenhang@kuaishou.com

## Related Issues
close #654

### CODE REVIEW

**Comment by aFlyBird0** [comment]
Thank you for your Contribution very much! I will review this PR soon. @azhsmesos

File: internal/pkg/pluginmanager/manager.go [hunk 1]
```diff
@@ -65,19 +65,16 @@ func DownloadPlugins(conf *configloader.Config) error {
...
-		// if existing .so doesn't matches with .md5, re-download
-		log.Infof("Plugin: %s doesn't match with .md5 and will be downloaded.", pluginFileName)
-		if err := redownloadPlugins(dc, pluginDir, pluginFileName, pluginMD5FileName, version.Version); err != nil {
-			return err
-		}
-
-		// check if the downloaded plugin md5 matches with .md5
-		if err := pluginAndMD5Matches(pluginDir, pluginFileName, pluginMD5FileName, tool.Name); err != nil {
-			return err
+		if !isMD5Match {
```
Comment (aFlyBird0) [inline internal/pkg/pluginmanager/manager.go:68]: Good! I think it does work.

File: internal/pkg/pluginmanager/manager.go [hunk 2]
```diff
@@ -65,19 +65,16 @@ func DownloadPlugins(conf *configloader.Config) error {
 			return err
 		}
 		// if .so matches with .md5, continue
-		if isMD5Match {
```
Comment (aFlyBird0) [inline internal/pkg/pluginmanager/manager.go:68]: Perhaps we can simply modify it here. If this code is deleted, then the user will not receive the "plugin exists" message, even if the plugin has actually been downloaded before.

Here is a simple way to implement it: use three conditions in the meanwhile to check if the plugin is been downloaded before.

- condition 1: plugin exist
- condition 2: .md5 exist
- condition 3: isMD5Matc(it is exists in the code now)

### condition 1 :
https://github.com/devstream-io/devstream/blob/ddb7c165300de00d31db97b4c16121cbb7ada2d2/internal/pkg/pluginmanager/manager.go#L37-L50
else condition 1 is true

### condition2:
https://github.com/devstream-io/devstream/blob/ddb7c165300de00d31db97b4c16121cbb7ada2d2/internal/pkg/pluginmanager/manager.go#L51-L61
else condition 2 is true

note: you should not use `contion1` as var name. 🤣

**Comment by aFlyBird0** [comment]
btw, to make your first contribution easier and happier, there are some suggestions that may be useful for you. @azhsmesos 

1. You be you have noticed that the CI failed because of "DCO". That means you missed Sign Off in your commits. You can check this [Chinese doc about sign-off](https://docs.devstream.io/en/latest/contributing_guide.zh/#sign-off).  Also, you can use Goland to help you do that. 
image

2. You may be bothered by some small commits. I guess you really want to make them into one commit. You can use `git rebase` to do that. Or you can use Goland to do it and it is easier. https://blog.csdn.net/weixin_43753680/article/details/117363518

Here is  the suggested workflow:
1. Change the code under the guide of review.
2. Do not forget to Sign Off on each commit.
3. rebase commits.
4. execute `git push origin -f` to force push codes to here.

**Comment by aFlyBird0** [comment]
In addition, feat means a new feature.

So feat(bug) may not suitable.

I think you could write commit message in this way:

`fix: xxx`

or

`fix(pluginmanager):xxx`

Hope that I don't bother you. I comment so  much just because I want you complete this pr happier and easier.

Please feel free tk ask me for any help. :heart:

**Comment by aFlyBird0** [comment]
> 话说DCO这个，我不太清楚该怎么弄 image

Are you convenient to add my WeChat? just search aFlyBird0

**Comment by azhsmesos** [comment]
> > 话说DCO这个，我不太清楚该怎么弄 image
> 
> Are you convenient to add my WeChat? just search aFlyBird0

Already added

**Review by aFlyBird0 — APPROVED** [review]
Perfect! I think there is no problem now.

**Review by daniel-hutao — APPROVED** [review]

**Comment by IronCore864** [comment]
@azhsmesos Hi,

Thanks for your contribution! 

Please join our [Slack here](https://join.slack.com/t/devstream-io/shared_invite/zt-16tb0iwzr-krcFGYRN7~Vv1suGZjdv4w). Feel free to reach out and request to join the [contributor channel](https://devstream-io.slack.com/messages/contributors)! We are also in CNCF now; please also consider joining [this channel in CNCF Slack](https://cloud-native.slack.com/messages/devstream).

In the meantime, feel free to check out our [new contributor guide](https://docs.devstream.io/en/latest/contributing_guide/#new-contributor-guide). We have quite a few pieces of documentation explaining the basic concepts and how to work on DevStream. Read them in the ["Developer Guide"](https://docs.devstream.io/en/latest/development/development-workflow/) section of the documentation.

If you are searching for something else to work on, please do check out the ["find an issue"](https://docs.devstream.io/en/latest/contributing_guide/#find-an-issue) section. The ["help wanted"](https://github.com/devstream-io/devstream/labels/help%20wanted) label is an excellent place to continue. If you are not completely confident yet, maybe you can warm yourself up by solving another ["good first issue"](https://docs.devstream.io/en/latest/development/good-first-issues/).

**Pull request merged**